### PR TITLE
Adding implementation text for Grids

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

Adding implementation text for Grids

## Why is the change being made?

This is part of on-going QA effort.

---

## Checklist

- [X] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [X] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [X] If any [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were affected, mention it in the [release notes](https://terrapower.github.io/armi/release/index.html).
- [X] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/reactor/grids/hexagonal.py [hunk 1]
```diff
@@ -47,13 +47,20 @@ class HexGrid(StructuredGrid):
...
+    It is recommended to use :meth:`fromPitch` rather than calling the ``__init__``
+    constructor directly.
 
     .. impl:: Construct a hexagonal lattice.
         :id: I_ARMI_GRID_HEX
         :implements: R_ARMI_GRID_HEX
 
+        This class represents a hexagonal ``StructuredGrid``, that is one where the
+        mesh maps to real, physical coordinates. This hexagonal grid is 2D, and divides
+        the plane up into regular, hexagons. That is, each hexagon is symmetric and
```
Comment (opotowsky) [inline armi/reactor/grids/hexagonal.py:59]:
```suggestion
        the plane up into regular hexagons. That is, each hexagon is symmetric and
```

File: armi/reactor/grids/grid.py [hunk 2]
```diff
@@ -90,6 +99,21 @@ def symmetry(self) -> str:
         .. impl:: Grids shall be able to repesent 1/3 and full core symmetries.
             :id: I_ARMI_GRID_SYMMETRY
```
Comment (opotowsky) [inline armi/reactor/grids/grid.py:100]:
```suggestion
            :id: I_ARMI_GRID_SYMMETRY0
```
optional...I see a `1` down below

File: armi/reactor/grids/structuredgrid.py [hunk 3]
```diff
@@ -290,11 +290,23 @@ def restoreBackup(self):
...
+
+            Probably the most common request of a structure grid will be to give the
+            grid indices and return the physical coordinates of the center of the mesh
+            cell. This is super handy in any situation where the coordinates have
+            physical meaning.
+
+            The math for finding the centroid turns out to be very easy, as the mesh is
+            defined on the coordinates. So finding the mid-point along one axis is just
+            taking the upper and lower bounds and dividing by two. And this is done for
+            all axis. There are no more complicated situations where we need to find
```
Comment (opotowsky) [inline armi/reactor/grids/structuredgrid.py:308]:
```suggestion
            all axes. There are no more complicated situations where we need to find
```
?

File: armi/reactor/grids/grid.py [hunk 4]
```diff
@@ -39,6 +39,16 @@ class Grid(ABC):
         :id: I_ARMI_GRID_NEST
         :implements: R_ARMI_GRID_NEST
 
+        The reactor will usually have (i,j,k) coordinates to define a
```
[the remark this answers is not in the document]
Comment (john-science) [reply armi/reactor/grids/grid.py:42]: See what you think.

File: armi/reactor/grids/grid.py [hunk 5]
```diff
@@ -39,6 +39,19 @@ class Grid(ABC):
...
+        simple mesh for locating objects in the reactor. But inside that mesh can
+        be a smaller mesh to define the layout of pins in a reactor, or fuel pellets in
+        a pin, or the layout of some intricate ex-core structure.
+
+        Every time the :py:class:` armi.reactor.grids.locations.IndexLocation` of an
```
Comment (opotowsky) [inline armi/reactor/grids/grid.py:47]: is this space between ` and armi going to render OK?

File: armi/reactor/grids/grid.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (opotowsky) [reply armi/reactor/grids/grid.py:42]: love it, thanks!

**Review by opotowsky — APPROVED** [review]
Just a little potential formatting error I commented on, but I'll approve now since the content is good to go!

File: armi/reactor/grids/grid.py (hunk 5 above)
Comment (john-science) [inline armi/reactor/grids/grid.py:47]:
```suggestion
        Every time the :py:class:`armi.reactor.grids.locations.IndexLocation` of an
```

Comment (john-science) [reply armi/reactor/grids/grid.py]: No idea, so I fixed it.

**Pull request merged**